contracts/stats/StatBlock.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+/**
+ * Authors: Moonstream Engineering (engineering - at - moonstream.to)
+ * GitHub: https://github.com/moonstream-to/web3
+ */
+
+import {IERC1155} from "@openzeppelin-contracts/contracts/token/ERC1155/IERC1155.sol";
+import {IStatBlock} from "./IStatBlock.sol";
+
+contract StatBlock is IStatBlock {
+    address AdminTerminusAddress;
+    uint256 AdminTerminusPoolID;
+    // Stats are 0-indexed.
+    uint256 public NumStats;
+    mapping(uint256 => string) StatDescriptor;
+    mapping(address => mapping(uint256 => mapping(uint256 => uint256))) Stat;
+
+    constructor(address adminTerminusAddress, uint256 adminTerminusPoolID) {
+        AdminTerminusAddress = adminTerminusAddress;
+        AdminTerminusPoolID = adminTerminusPoolID;
+    }
+
+    function adminTerminusInfo() external view returns (address, uint256) {
+        return (AdminTerminusAddress, AdminTerminusPoolID);
+    }
+
+    function isAdministrator(address account) public view returns (bool) {
+        IERC1155 terminus = IERC1155(AdminTerminusAddress);
+        return terminus.balanceOf(account, AdminTerminusPoolID) > 0;
+    }
+
+    function createStat(
+        string memory descriptor
+    ) external returns (uint256 statID) {
+        require(
+            isAdministrator(msg.sender),
+            "StatBlock.createStat: msg.sender must be an administrator of the StatBlock"
+        );
+        statID = NumStats++;
+        StatDescriptor[statID] = descriptor;
+        emit StatCreated(statID, descriptor);
+    }
+
```

Comment:
I think we should add a setStatDescriptor method. Feels a bit brittle otherwise.